balena/models/device_os.py

```diff
@@ -107,6 +119,49 @@ def download(self, raw=None, **data):
         else:
             return response
 
+    def download_unconfigured_image(self, device_type, version='latest', raw=False):
+        """
+        Download an unconfigured OS image. This function only works if you log in using credentials or Auth Token.
+
+        Args:
+            device_type (str): device type slug.
+            version (str): the Balena OS version of the image (defaults to 'latest'). The SDK will try to parse version into semver-compatible version, unsupported (unpublished) version will result in rejection.
+            raw (bool): determining function return value.
+
+        Returns:
+            object:
+                If raw is True, urllib3.HTTPResponse object is returned.
+                If raw is False, original response object is returned.
+
+        Notes:
+            default OS image file name can be found in response headers.
+
+        Examples:
+            >>> response = balena.models.device_os.download_unconfigured_image('raspberry-pi2', 'latest')
+            >>> type(response)
+            <class 'requests.models.Response'>
+            >>> response['headers']
+            >>> response.headers
+            {'Access-Control-Allow-Headers': 'Content-Type, Authorization, Application-Record-Count, MaxDataServiceVersion, X-Requested-With, X-Balena-Client', 'content-disposition': 'attachment; filename="balena-cloud-raspberry-pi2-2.43.0+rev1-v10.2.2.img"', 'X-Content-Type-Options': 'nosniff', 'Access-Control-Max-Age': '86400', 'Transfer-Encoding': 'chunked', 'x-powered-by': 'Express', 'content-encoding': 'gzip', 'x-transfer-length': '134445269', 'last-modified': 'Mon, 23 Sep 2019 15:21:33 GMT', 'Connection': 'keep-alive', 'Access-Control-Allow-Credentials': 'true', 'Date': 'Tue, 07 Jan 2020 18:14:47 GMT', 'X-Frame-Options': 'DENY', 'Access-Control-Allow-Methods': 'GET, PUT, POST, PATCH, DELETE, OPTIONS, HEAD', 'Content-Type': 'application/octet-stream', 'Access-Control-Allow-Origin': '*'}
+
+        """
+
+        if version == 'latest':
+            version = self.get_supported_versions(device_type)['latest']
+        else:
+            version = self.get_device_os_semver_with_variant(version, 'prod')
```

Comment:
I think that we also support downloading uncofigured dev images. As suggested above, how about inferring the `dev`/`prod` variant using a `/\bdev\b/` regex?

Reply:
Updated (The SDK already has the regex to handle dev variant here https://github.com/balena-io/balena-sdk-python/blob/master/balena/models/device_os.py#L155-L169)